graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -164,16 +164,16 @@ c10::intrusive_ptr<sampling::FusedSampledSubgraph> SampleNeighbors(
   auto max_in_degree = torch::empty(
       1,
       c10::TensorOptions().dtype(in_degree.scalar_type()).pinned_memory(true));
-  AT_DISPATCH_INTEGRAL_TYPES(
+  AT_DISPATCH_INDEX_TYPES(
```

Comment:
the main advantage of this new macro is only generating code for `int32_t` and `int64_t` only, instead of all possible integral data types? so less code, less compile time, less compiled lib size?

Reply:
Exactly. When we have nested dispatches, the combinations become exponential.